### PR TITLE
Enh deploy release v6

### PR DESCRIPTION
**Pull request by benPearce1**
```
- task: OctopusDeployRelease@6
  name: 'release'
  inputs:
    OctoConnectedServiceName: 'octo'
    Space: 'Default'
    Project: 'project 1'
    ReleaseNumber: '1.2.10'
    Environments: 'dev,dev 2'
    Variables: |
      prompt: $(build.buildid)
    AdditionalArguments: --variable prompt2=prompt2
```

Supports variables coming from either additional arguments or the variable field

In ui mode
image

### CODE REVIEW

File: pack.ps1 [hunk 1]
```diff
@@ -127,11 +126,12 @@ function Pack($envName, $environment, $workingDirectory) {
 
     & ./node_modules/.bin/tfx extension create --root $workingDirectory --manifest-globs extension-manifest.json --overridesFile $overridesFile --outputPath "$buildArtifactsPath/$environment" --no-prompt
 
-    if (-not $?) {throw "Failed to create extension. Exit Code $LASTEXITCODE"}
+    & npm uninstall tfx-cli
```
Comment (benPearce1) [inline pack.ps1:129]: This is the only change to this line, the rest was auto-formatting kicking in

File: source/tasks/Deploy/DeployV6/input-parameters.test.ts [hunk 2]
```diff
@@ -0,0 +1,104 @@
+import { Logger } from "@octopusdeploy/api-client";
+import { TaskWrapper } from "tasks/Utils/taskInput";
+import { getInputParameters } from "./input-parameters";
+import * as tasks from "azure-pipelines-task-lib/task";
+
+export class MockTaskWrapper implements TaskWrapper {
```
Comment (benPearce1) [inline source/tasks/Deploy/DeployV6/input-parameters.test.ts:6]: This needs to move to a common location, but I initially had a reference issue so I put it here for now

File: source/tasks/Deploy/DeployV6/task.json [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+            "required": false,
+            "helpMarkDown": "Whether to use guided failure mode if errors occur during the deployment."
+        },
+        {
+            "name": "AdditionalArguments",
+            "type": "string",
+            "label": "Additional Arguments (deprecated)",
+            "defaultValue": "",
+            "required": false,
+            "helpMarkDown": "Additional arguments are no longer supported, move these values to the appropriate fields",
```
Comment (benPearce1) [inline source/tasks/Deploy/DeployV6/task.json:102]: Wording?

File: source/tasks/Deploy/DeployV6/task.json [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+            "groupName": "advanced"
+        }
+    ],
+    "instanceNameFormat": "Deploy Octopus Release",
+    "execution": {
+        "Node16": {
```
Comment (benPearce1) [inline source/tasks/Deploy/DeployV6/task.json:108]: updated

File: source/tasks/Utils/command-line-args.test.ts [hunk 5]
```diff
@@ -0,0 +1,52 @@
+import commandLineArgs from "command-line-args";
+import shlex from "shlex";
+
+describe("cli args", () => {
```
Comment (benPearce1) [inline source/tasks/Utils/command-line-args.test.ts:4]: These tests were me working out how to parse the arguments from the `AdditionalArguments` field, thought I would leave them here for future reference

File: source/tasks/Deploy/DeployV6/task.json (hunk 4 above)
Comment (slewis74) [reply source/tasks/Deploy/DeployV6/task.json:108]: 👍

File: source/tasks/Deploy/DeployV6/task.json [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+        {
+            "name": "Environments",
+            "type": "multiLine",
+            "label": "Deploy to Environments",
+            "defaultValue": "",
+            "required": true,
+            "properties": {
+                "EditableOptions": "True"
+            },
+            "helpMarkDown": "List of environments to deploy to, one environment per line"
```
Comment (slewis74) [inline source/tasks/Deploy/DeployV6/task.json:75]:
```suggestion
            "helpMarkDown": "List of environments to deploy to, one environment per line. (Note: multiple on a line, separated by commas, is supported to ease migration from earlier versions of the step but this format should be considered deprecated)"
```

File: source/tasks/Deploy/DeployV6/task.json [hunk 7]
```diff
@@ -0,0 +1,112 @@
...
+            "required": false,
+            "helpMarkDown": "Whether to use guided failure mode if errors occur during the deployment."
+        },
+        {
+            "name": "AdditionalArguments",
+            "type": "string",
+            "label": "Additional Arguments (deprecated)",
+            "defaultValue": "",
+            "required": false,
+            "helpMarkDown": "Additional arguments are no longer supported, move these values to the appropriate fields",
```
Comment (slewis74) [inline source/tasks/Deploy/DeployV6/task.json:102]:
```suggestion
            "helpMarkDown": "Additional arguments are no longer supported. This field has been retained to ease migration from earlier versions of the step but values should be moved to the appropriate fields.",
```

**Review by slewis74 — CHANGES_REQUESTED** [review]
Looking good. Offered some ideas on the wording for the no longer supported fields. `vsts.md` hasn't been updated.

**Review by slewis74 — APPROVED** [review]

**Pull request merged**